tests/antithesis/test-template/Dockerfile

```diff
@@ -0,0 +1,26 @@
+ARG GO_VERSION=1.24.2
+ARG ARCH=amd64
+
+FROM golang:$GO_VERSION AS build
+
+# Copy replace directives for build
+# TODO - there should be a way to automate this
+ARG BUILD_DIR=/building
+
+WORKDIR $BUILD_DIR
+COPY . .
+
+# Run go mod tidy to ensure the dependencies are correct
+RUN go mod tidy
+
+# Building entrypoint 
```

Comment:
`WORKDIR $BUILD_DIR/tests`? This would avoid the cd which reduces readability.

Reply:
Done!